subsys/bluetooth/services/fast_pair/fp_crypto/fp_crypto_common.c

```diff
@@ -130,23 +130,30 @@ size_t fp_crypto_account_key_filter_size(size_t n)
 }
 
 int fp_crypto_account_key_filter(uint8_t *out, const struct fp_account_key *account_key_list,
-				 size_t n, uint8_t salt)
+				 size_t n, uint8_t salt, const uint8_t *battery_info)
 {
 	size_t s = fp_crypto_account_key_filter_size(n);
-	uint8_t v[FP_ACCOUNT_KEY_LEN + sizeof(salt)];
+	uint8_t v[FP_ACCOUNT_KEY_LEN + sizeof(salt) + FP_CRYPTO_BATTERY_INFO_LEN];
 	uint8_t h[FP_CRYPTO_SHA256_HASH_LEN];
 	uint32_t x;
 	uint32_t m;
+	int err;
 
 	memset(out, 0, s);
 	for (size_t i = 0; i < n; i++) {
 		memcpy(v, account_key_list[i].key, FP_ACCOUNT_KEY_LEN);
 		v[FP_ACCOUNT_KEY_LEN] = salt;
-		int err = fp_crypto_sha256(h, v, sizeof(v));
-
+		if (battery_info != NULL) {
```

Comment:
Maybe we could introduce a `size_t pos` value and use it as current array index? That would simplify code here (data size for SHA256 would be automatically calculated too)